packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts

```diff
@@ -0,0 +1,79 @@
+import { useState, useEffect } from "react";
+import { Environment, AxelarQueryAPI } from "@axelar-network/axelarjs-sdk";
+import { CoinPretty } from "@keplr-wallet/unit";
+import { AppCurrency } from "@keplr-wallet/types";
+
+/** Fetches a new Transfer fee quote when either chain, the amount, or the currency changes.
+ *  `amountMinDenom` is from user input, assumes `=== ""` for no input, and therefore no query.
+ */
+export function useTransferFeeQuery(
+  sourceChain: string,
+  destChain: string,
+  tokenMinDenom: string,
+  amountMinDenom: string,
+  currency: AppCurrency,
+  environment = Environment.MAINNET,
+  inputDebounceMs = 1400
+): { transferFee?: CoinPretty; isLoading: boolean } {
+  const [transferFee, setTransferFee] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const queryTransferFee = async () => {
+      const api = new AxelarQueryAPI({
+        environment,
+      });
+      const amount = Number(
+        new CoinPretty(
+          currency,
+          amountMinDenom === "" ? "0" : amountMinDenom
+        ).toCoin().amount
+      );
+      if (!isNaN(amount)) {
+        console.log("amount " + amount);
+        return await api.getTransferFee(
+          sourceChain,
+          destChain,
+          tokenMinDenom,
+          amount
+        );
+      } else {
+        throw new Error("Requested fee amount is not a number.");
+      }
+    };
+    let timeout: NodeJS.Timeout | undefined;
+
+    setIsLoading(true);
+    // debounce query on user typing
+    timeout = setTimeout(() => {
```

Comment:
Could you please create a general debounce hook in the /hooks/ folder? Note- you may be able to leverage a `debounce` package we already download into our bundle. It should be able to watch any given value and only run an effect once some debounce period has elapsed.